apps/patient/src/components/ReadyText.tsx

```diff
@@ -38,47 +34,70 @@ export const ReadyText = ({
   return null;
 };
 
-const roundUpTo15MinInterval = (pharmacyEstimatedReadyAt: Date): Dayjs => {
-  const estimatedReadyAtAsDayJs = dayjs(pharmacyEstimatedReadyAt);
-  const minutes = estimatedReadyAtAsDayJs.minute();
+const roundUpTo15MinInterval = (pharmacyEstimatedReadyAt: Date): Date => {
+  const estimatedReadyAt = new Date(pharmacyEstimatedReadyAt);
+  const minutes = estimatedReadyAt.getMinutes();
   const roundedMinutes = Math.ceil(minutes / 15) * 15;
+
   if (roundedMinutes >= 60) {
-    return estimatedReadyAtAsDayJs.add(1, 'hour').minute(0);
+    estimatedReadyAt.setHours(estimatedReadyAt.getHours() + 1);
+    estimatedReadyAt.setMinutes(0);
   } else {
-    return estimatedReadyAtAsDayJs.minute(roundedMinutes);
+    estimatedReadyAt.setMinutes(roundedMinutes);
   }
+
+  estimatedReadyAt.setSeconds(0);
+  estimatedReadyAt.setMilliseconds(0);
+  return estimatedReadyAt;
 };
 
 interface PharmacyEstimatedReadyAtProps {
   pharmacyEstimatedReadyAt: Date;
 }
 const PharmacyEstimatedReadyAt = ({ pharmacyEstimatedReadyAt }: PharmacyEstimatedReadyAtProps) => {
-  const readyAt = roundUpTo15MinInterval(pharmacyEstimatedReadyAt);
-  const isTomorrow = readyAt.add(1, 'day').isTomorrow();
-  const timeFormat = readyAt.minute() ? 'h:mm a' : 'h a';
+  const rounded = roundUpTo15MinInterval(pharmacyEstimatedReadyAt);
+  const timezone = dayjs.tz.guess();
+  const readyAtDayjs = dayjs.utc(rounded).tz(timezone);
+  const timeFormat = readyAtDayjs.minute() ? 'h:mm a' : 'h a';
+  const now = dayjs();
+  const isFuture = now.isBefore(readyAtDayjs);
 
-  return (
-    <Text>
-      Ready{isTomorrow ? <b> Tomorrow </b> : ' '} at <b>{readyAt.format(timeFormat)}</b>
-    </Text>
-  );
+  if (readyAtDayjs.isToday() && isFuture) {
```

Comment:
I added this future check because if it isToday but it's past, i think "ready at 10 am" could be mistaken for tomorrow at 10 am